app/models/play_group.rb

```diff
@@ -0,0 +1,192 @@
+class PlayGroup
+  def initialize(params)
+    @error = nil
+    @group_sizes = set_group_sizes(params)
+    pp @group_sizes
+    @players = set_players(params)
+    @collection = set_collection(params)
+    @collection.present? ? @games = set_games(params) : @error = "No rated games found, please add games or add players with existing collections, or go and rate some games!"
+    params[:group_size_max] = player_count if player_count < params[:group_size_min].to_i
+    @error = "Group sizes not achievable with player count, please either amend group sizes or add or remove players" if group_error_check(params)
+    if !@error.present?
+      @groups = set_groups(params)
+      @group_combinations = combine_groups(params)
+      @player_scores = set_player_scores
+      @group_scores = calculate_group_scores(params)
+      pp " scores: #{@group_scores}"
+      !@group_scores[0].first[1].empty? ? @final_scores = calculate_final_scores(params) : @error = "No rated games meet criteria, please adjust options and try again, or go and rate some games!"
+      end      
+  end
+
+  def nested_hash
+    Hash.new { |h, k| h[k] = nested_hash }
+  end
+
+  def get_groups 
+    @groups
+  end
+
+  def get_final_scores
+    @final_scores
+  end
+
+  def get_players
+    @players
+  end
+
+  def get_games
+    @games
+  end
+
+  def get_scores
+    @player_scores
+  end
+
+  def get_group_rankings(group)
+    @group_scores[group]
+  end
+
+  def get_group_scores
+    @group_scores
+  end
+
+  def get_group_score(group, game)
+    @group_scores[group][game]
+  end
+
+  def set_players(params)
+    user_ids = params.select{ |k, v| (k.start_with? "attend") && (v == "1") }.keys.map{|k| k.tr("attend","").to_i}
+    User.where(id: user_ids)
+  end
+
+  def get_error
+    @error
+  end
+
+  def calculate_group_scores(params)
+    scores = groups_map(params)
+  end
+
+  def groups_map(params)
+    @groups.map{|group| {group => games_map(group,params)}}
+  end
+
+  def games_map(group,params)
+    ratings = Rating.where user_id: group, game_id: @games
```

Comment:
Similar to `set_player_scores`, get all ratings up front and pass through to the methods that use them to avoid querying the table a lot